common/src/main/java/io/netty/util/internal/ObjectCleaner.java

```diff
@@ -72,6 +71,7 @@ public void run() {
                 if (LIVE_SET.isEmpty() || !CLEANER_RUNNING.compareAndSet(false, true)) {
                     // There was nothing added after we set STARTED to false or some other cleanup Thread
                     // was started already so its safe to let this Thread complete now.
+                    ObjectCleaner.objectCleanerThread = null;
```

Comment:
This may race with the thread setting the variable in the register method below.